### PR TITLE
Support files without extensions / hidden files

### PR DESCRIPTION
**Pull request by mkurz**
First I push a commit to update the current simple scripted test to see if my coming changes will not break backwards compatibility.

Update: That first commit [passed the tests](https://github.com/sbt/sbt-header/actions/runs/3272840586/jobs/5384329716)  :+1: 
Now I pushed the real "fix" and tests for it.

### CODE REVIEW

File: src/sbt-test/sbt-header/no-file-extensions/test.sbt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+headerLicense := Some(HeaderLicense.ALv2("2015", "Heiko Seeberger"))
+
+// Files named exactly "routes" (with no extension)
+headerMappings := headerMappings.value + (HeaderFileType("", None, "routes") -> HeaderCommentStyle.hashLineComment)
+// Files named exactly "more.routes"
+headerMappings := headerMappings.value + (HeaderFileType("routes", None, "more") -> HeaderCommentStyle.cStyleBlockComment)
+// Hidden files named exactly ".routes"
+headerMappings := headerMappings.value + (HeaderFileType("", None, ".routes") -> HeaderCommentStyle.cppStyleLineComment)
```
Comment (mkurz) [inline src/sbt-test/sbt-header/no-file-extensions/test.sbt:8]: I think these three lines explains what this whole pull request is all about.

**Comment by mkurz** [comment]
@hseeberger Can you take a look? This pull request is fully compatible with previous releases (see the first commit I added) and we would need that feature in the Play Framework. Thanks!

**Comment by hseeberger** [comment]
Looking good. Thanks a lot!

**Comment by mkurz** [comment]
@hseeberger Can you please also cut a new release? Thanks!

**Comment by mkurz** [comment]
Thanks for the release!

**Pull request merged**